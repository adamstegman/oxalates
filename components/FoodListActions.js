import React from 'react';
import { Button, StyleSheet, TextInput, View } from 'react-native';

import Colors from '../constants/Colors';

export class FoodListActions extends React.Component {
  constructor() {
    super();
    this.state = { admin: false };
  }

  componentDidMount() {
    this.setState({ admin: this.props.admin });
  }

  logIn() {
    this.setState({ admin: true });
  }

  logOut() {
    this.setState({ admin: false });
  }

  render() {
    let sessionButton;
    if (this.state.admin) {
      sessionButton = <Button
        title="Log Out"
        onPress={() => this.logOut()}
      />;
    } else {
      sessionButton = <Button
        title="Log In"
        onPress={() => this.logIn()}
      />;
    }

    const searchContainerStyle = StyleSheet.flatten([styles.actionContainer, styles.searchContainer]);
    return (
      <View style={styles.container}>
        <View style={styles.actionContainer}>
          {sessionButton}
        </View>
        <View style={searchContainerStyle}>
          <TextInput
            autoCapitalize={"none"}
            clearButtonMode={"always"}
            placeholder={"Filter"}
            returnKeyType={"search"}
            style={styles.search}
            onChangeText={query => this.props.onFilter(query)}
          />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.listHeaderBackgroundColor,
    flexDirection: 'row',
    justifyContent: 'space-between',
    height: 48,
  },
  actionContainer: {
    flex: 1,
    margin: 4,
  },
  searchContainer: {
    borderRadius: 8,
    overflow: 'hidden',
  },
  search: {
    backgroundColor: Colors.backgroundColor,
    padding: 10,
    fontSize: 16,
  },
});
