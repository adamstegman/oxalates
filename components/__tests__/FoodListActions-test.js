import 'react-native';
import React from 'react';
import { FoodListActions } from '../FoodListActions';
import renderer from 'react-test-renderer';

it('renders the food list actions', () => {
  const tree = renderer.create(<FoodListActions />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders the food list actions for an admin', () => {
  const tree = renderer.create(<FoodListActions admin={true} />).toJSON();

  expect(tree).toMatchSnapshot();
});
