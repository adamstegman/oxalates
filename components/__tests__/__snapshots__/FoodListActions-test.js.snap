// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the food list actions 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#d8d8d8",
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "margin": 4,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Log In
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 8,
        "flex": 1,
        "margin": 4,
        "overflow": "hidden",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Filter"
      returnKeyType="search"
      style={
        Object {
          "backgroundColor": "#fff",
          "fontSize": 16,
          "padding": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`renders the food list actions for an admin 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#d8d8d8",
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "margin": 4,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Log Out
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 8,
        "flex": 1,
        "margin": 4,
        "overflow": "hidden",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      clearButtonMode="always"
      onChangeText={[Function]}
      placeholder="Filter"
      returnKeyType="search"
      style={
        Object {
          "backgroundColor": "#fff",
          "fontSize": 16,
          "padding": 10,
        }
      }
    />
  </View>
</View>
`;
